apps/crn-server/src/utils/analytics/collaboration.ts

```diff
@@ -0,0 +1,160 @@
+import { FetchUserCoproductionQuery } from '@asap-hub/contentful';
+import { TeamRole, UserCollaborationDataObject } from '@asap-hub/model';
+import { cleanArray, parseUserDisplayName } from '@asap-hub/server-common';
+
+export type EntityWithId = {
+  sys: {
+    id: string;
+  };
+};
+
+export type Author = {
+  id: string;
+  teams: EntityWithId[];
+  labs: EntityWithId[];
+};
+
+export const checkDifferentTeams = (
+  referenceTeams: EntityWithId[],
+  testTeams: EntityWithId[],
+): boolean =>
+  referenceTeams.every((referenceTeam) =>
+    testTeams.every((testTeam) => testTeam.sys.id !== referenceTeam.sys.id),
+  );
+
+export const checkSameTeamDifferentLab = (
+  referenceTeam: EntityWithId,
+  referenceLabs: EntityWithId[],
+  testUser: Author,
+): boolean => {
+  const sameTeam = testUser.teams.find(
+    (testTeam) => testTeam.sys.id === referenceTeam.sys.id,
+  );
+  if (sameTeam) {
+    return referenceLabs
+      .map((lab) =>
+        testUser.labs.find((testLab) => testLab.sys.id === lab.sys.id),
+      )
+      .every((value) => !value);
+  }
+  return false;
+};
+
+export const findMatchingAuthors = ({
+  referenceId,
+  referenceTeams,
+  referenceTeam,
+  referenceLabs,
+  authorList,
+}: {
+  referenceId: string;
+  referenceTeam: EntityWithId;
+  referenceTeams: EntityWithId[];
+  referenceLabs: EntityWithId[];
+  authorList: Author[];
+}) => {
+  let [differentTeamFlag, sameTeamDifferentLabFlag] = [false, false];
+  let isAuthor = false;
+  authorList.forEach((author) => {
+    if (author && author.id !== referenceId) {
+      if (
+        !differentTeamFlag &&
+        checkDifferentTeams(referenceTeams, author.teams)
+      ) {
+        differentTeamFlag = true;
+      }
+      if (
+        !sameTeamDifferentLabFlag &&
+        checkSameTeamDifferentLab(referenceTeam, referenceLabs, author)
+      ) {
+        sameTeamDifferentLabFlag = true;
+      }
+    } else if (author.id === referenceId) {
+      isAuthor = true;
+    }
+  });
+  if (isAuthor) {
+    return { differentTeamFlag, sameTeamDifferentLabFlag };
+  }
+  // don't count output if not the author
+  return { differentTeamFlag: false, sameTeamDifferentLabFlag: false };
+};
+
+export const getCollaborationCounts = (
+  data: {
+    differentTeamFlag: boolean;
+    sameTeamDifferentLabFlag: boolean;
+  }[],
+) =>
+  data.reduce(
+    (
+      { acrossTeamCount, withinTeamCount },
+      { differentTeamFlag, sameTeamDifferentLabFlag },
+    ) => ({
+      acrossTeamCount: differentTeamFlag
+        ? acrossTeamCount + 1
+        : acrossTeamCount,
+      withinTeamCount: sameTeamDifferentLabFlag
+        ? withinTeamCount + 1
+        : withinTeamCount,
+    }),
+    {
+      acrossTeamCount: 0,
+      withinTeamCount: 0,
+    },
+  );
+
+export const getUserCoProductionItems = (
+  userCollection: FetchUserCoproductionQuery['usersCollection'],
```

Comment:
```suggestion
  userCollection: FetchUserCoProductionQuery['usersCollection'],
```

Reply:
this is coming from autogen code, don't think I can rename it without making it worse. but I don't like the name to be honest, I think I'll rename to collaboration and make it consistent everywhere.